pelican/plugins/sitemap/sitemap.py

```diff
@@ -55,235 +47,177 @@ def format_date(date):
-            + self.context["articles"]
-            + [c for (c, a) in self.context["categories"]]
-            + [t for (t, a) in self.context["tags"]]
-            + [a for (a, b) in self.context["authors"]]
+        page_queue = [(to_url(path), obj) for path, obj in self.page_queue]
+        # Clean after filtering, because that was the way before, sort
+        page_queue = sorted(
+            (clean_url(url), obj) for url, obj in filterfalse(is_excluded, page_queue)
         )
 
-        self.set_url_wrappers_modification_date(self.context["categories"])
-        self.set_url_wrappers_modification_date(self.context["tags"])
-        self.set_url_wrappers_modification_date(self.context["authors"])
-
-        for article in self.context["articles"]:
-            pages += article.translations
-
-        info("writing {0}".format(path))
-
-        with open(path, "w", encoding="utf-8") as fd:
-
-            if self.format == "xml":
+        with open(filename, "w", encoding="utf-8") as fd:
+            if is_xml:
                 fd.write(XML_HEADER)
-            else:
-                fd.write(TXT_HEADER.format(self.siteurl))
 
-            FakePage = collections.namedtuple(
-                "FakePage", ["status", "date", "url", "save_as"]
-            )
+            for pageurl, obj in page_queue:
+                if not is_xml:
+                    fd.write(siteurl + "/" + pageurl + "\n")
+                    # That's it for txt. Short circuit the loop, gain an indent level.
+                    continue
 
-            for standard_page in self.context["DIRECT_TEMPLATES"]:
-                standard_page_url = self.context.get(
-                    "{}_URL".format(standard_page.upper())
+                lastmod = format_date(
+                    getattr(obj, "modified", None)
+                    or getattr(obj, "date", None)
+                    or self.now
                 )
-                standard_page_save_as = self.context.get(
-                    "{}_SAVE_AS".format(standard_page.upper())
+                content_type = (
+                    "articles"
+                    if isinstance(obj, contents.Article)
+                    else "pages"
+                    if isinstance(obj, contents.Page)
+                    else "indexes"
                 )
-                fake = FakePage(
-                    status="published",
-                    date=self.now,
-                    url=standard_page_url or "{}.html".format(standard_page),
-                    save_as=standard_page_save_as or "{}.html".format(standard_page),
+                changefreq = changefreqs[content_type]
+                priority = float(priorities[content_type])
+                translations = "".join(
+                    XML_TRANSLATION.format(
+                        trans.lang,
+                        siteurl,
+                        # save_as path is already output-relative
+                        clean_url(pathname2url(trans.save_as)),
```

Comment:
This now works with i18n_subsites (:tada:), but here the produced URL in translation links is incorrect. `trans.save_as` is formatted according to e.g. `PAGE_LANG_URL` for all subsites (including the `DEFAULT_LANG` subsite where `PAGE_URL` is the one appropriate). 

Ideas on how else to get the proper final URL/path on disk?

Reply:
Can someone else confirm this? _i18n_subsites_ no longer build with my project. :grimacing: :joy: